From 8020f4b3c1b347054174177aa5cc1cec0f0ed25e Mon Sep 17 00:00:00 2001
From: Kumar Gala <kumar.gala@linaro.org>
Date: Thu, 14 Feb 2019 23:34:19 -0600
Subject: [PATCH] sys/_default_fcntl.h: workaround arc nsim build issue

If we are building newlib and have MISSING_SYSCALL_NAMES defined we run
into an issue when libgloss/arc/nsim-syscalls.c tries to build.  As
MISSING_SYSCALL_NAMES will cause _open to become open, than we can a
build conflict between how open is declared in sys/_default_fcntl.h and
how its defined in nsim-syscalls.c.

To workaround this we wrap the open declaration with
	#if !defined(MISSING_SYSCALL_NAMES)
	#endif

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
---
 newlib/libc/include/sys/_default_fcntl.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/newlib/libc/include/sys/_default_fcntl.h b/newlib/libc/include/sys/_default_fcntl.h
index 22fa10688..9466cb1b8 100644
--- a/newlib/libc/include/sys/_default_fcntl.h
+++ b/newlib/libc/include/sys/_default_fcntl.h
@@ -205,7 +205,9 @@ struct eflock {
 #include <sys/types.h>
 #include <sys/stat.h>		/* sigh. for the mode bits for open/creat */
 
+#if !defined(MISSING_SYSCALL_NAMES)
 extern int open (const char *, int, ...);
+#endif
 #if __ATFILE_VISIBLE
 extern int openat (int, const char *, int, ...);
 #endif
-- 
2.20.1

